spec/unit/provider/package/puppet_gem_spec.rb

```diff
@@ -67,6 +67,14 @@
       expect(described_class).to receive(:execute_gem_command).with(provider_gem_cmd, %w{uninstall --executables --all myresource --force --bindir=/usr/bin}).and_return('')
       provider.uninstall
     end
+
+    it 'should invalidate the rubygems cache' do
+      @gem_source = double('gem_source')
```

Comment:
`gem_source` can be a local variable

Reply:
To @gimmyxd point, we generally use local variables or use `let` to memoize a variable fpr each rspec test. Unfortunately there are lots of tests that still use `@name`, so it's understandably confusing.

Reply:
Changed to local var.